synapsebridgehelpers/tableStats.py

```diff
@@ -0,0 +1,179 @@
+#%matplotlib inline (This is for ipython notebook)
+
+import synapseclient
+import pandas as pd
+import numpy as np
+from collections import Counter
+import matplotlib.pylab as plt
+from datetime import datetime
+
+###########################################################################################
+
+def healthCodeRecords(df, returnType = 'series'):
+    """returns number of records per healthCode """
+    try:
+        sortedSeries = df['healthCode'].value_counts()        
+        if returnType == 'series':
+            return sortedSeries
+        else:
+            return dict(sortedSeries)
+    except:
+        print('The given dataframe does not have a column by that name')    
+        
+###########################################################################################
+
+def recordsVsHealthCodes(df):
+    """returns two lists 'nrecords' and 'nhealthcodes' which are number of records and
+    the corresponding number of healthCodes with that many records"""
+    try:
+        healthCodes = (df['healthCode'].value_counts())
+        countedHealthCodes = dict(Counter(healthCodes))
+        x,y = zip(*sorted(countedHealthCodes.items()))
+        return {'nrecords':x, 'nhealthcodes':y}
+    except:
+        print('The given dataframe does not have the column healthCodes')
+        
+###########################################################################################        
+        
+def plotRecordsVsHealthCodes(syn, table_id, nbins = 10, scale = 'linear'):
+    """Plots the number of records vs the number of healthcodes that have that 
+    many records"""
+    try:
+        results = syn.tableQuery('SELECT * FROM ' + table_id)
+        df =  results.asDataFrame()
```

Comment:
I wrote a function called summaryTable that does exactly that similar to the parkinson-appVersion. Thought it might come in handy, we can then give that tableId as input to this code